src/streamlink/stream/hls_playlist.py

```diff
@@ -239,15 +239,14 @@ def parse(self, data):
         self.state = {}
         self.m3u8 = M3U8()
 
-        lines = (l.rstrip() for l in StringIO(data.encode('utf-8')))
-        line = next(lines)
+        lines = enumerate(filter(bool, data.splitlines()))
```

Comment:
You don't need to use `enumerate` here if you are just ignoring the enumeration :)